src/electronic-ids/pcsc/FinEID.cpp

```diff
@@ -175,6 +180,9 @@ ElectronicID::PinRetriesRemainingAndMax FinEIDv3::pinRetriesLeft(byte_type pinRe
         THROW(SmartCardError,
               "Command GET DATA failed with error " + pcsc_cpp::bytes2hexstr(response.toBytes()));
     }
+    if (response.data.size() < 21) {
+        THROW(SmartCardError, "Command GET DATA failed with unexpected result data ");
```

Comment:
I think the error message should indicate what went wrong specifically - the size of the received data is less than the expected 21 bytes:

```suggestion
        THROW(SmartCardError, "Command GET DATA failed: received data size " + std::to_string(data.size()) + " is less than the expected 21 bytes");
```

Reply:
Im not sure that 21 is expected. It can be more